app/components/shared/card_component.rb

```diff
@@ -0,0 +1,59 @@
+class Shared::CardComponent < GovukComponent::Base
+  include ViewComponent::SlotableV2
+
+  renders_many :action_items, "ActionItemComponent"
+  renders_many :body_items, "BodyItemComponent"
+  renders_many :header_items, "HeaderItemComponent"
+
+  attr_accessor :html_attributes, :id
+
+  def initialize(id: nil, classes: [], html_attributes: {})
+    super(classes: classes, html_attributes: html_attributes)
+    @id = id
+  end
+
+  class HeaderItemComponent < ViewComponent::Base
```

Comment:
I don't have a problem with it, but why do you prefer to make these parts into component classes rather than methods?